corehq/apps/hqadmin/tests/test_pillowtop.py

```diff
@@ -52,22 +55,21 @@ def test_with_blacklist_items(self):
         self.assertEqual(expected_pillows, set([c.name for c in configs_back]))
 
     def test_loading_existing_conf_file(self):
+        config = {'include_groups': ['mvp_indicators']}
 
-        expected_action = {'include_groups': ['mvp_indicators']}
-
-        action = get_single_pillow_action('staging')
-        self.assertEqual(action.to_json(), expected_action)
-
-    def test_loading_no_existing_conf_file(self):
-        action = get_single_pillow_action('foo')
-        self.assertIsNone(action)
+        action = get_single_pillow_action(config)
+        self.assertEqual(action.to_json(), config)
 
     def test_india_server_exclusions(self):
         self.pillowtops['fluff'] = [
             'custom.bihar.models.CareBiharFluffPillow',
             'custom.opm.models.OpmUserFluffPillow',
         ]
+        configs = [
+            {'exclude_groups': ['mvp_indicators', 'fluff']},
+            {'include_pillows': ['custom.bihar.models.CareBiharFluffPillow']},
+        ]
 
-        pillows = [c.name for c in get_pillows_for_env('softlayer', self.pillowtops)]
+        pillows = [c.name for c in get_pillows_for_env(configs, self.pillowtops)]
         self.assertNotIn('OpmUserFluffPillow', pillows)
         self.assertIn('CareBiharFluffPillow', pillows)
```

Comment:
thanks for helping with this. i think of those 3 options you mention i like the last one the best. kind of have a dislike towards submodule, but suppose in circumstances where it truly is a good abstraction it makes sense. don't think i'll be tackling it soon, though it does seem like a worthwhile refactor to do